### DISCUSSION TITLE
What can it do ? Can it run games?

### DISCUSSION THREAD

**Discussion by matgaw123** [EduApps-CDG/OpenDX#11] [General]
What can I do with it ?

**Comment by EduApps-CDG** [comment]
It depends on what you meant by 'Can it run games?'. OpenDX implements functions like [`IDirect3D9::IDirect3D9(UINT SDKVersion)`](https://github.com/EduApps-CDG/OpenDX/blob/main/libs/d3d9/d3d9.cpp#L15) (which is only available on Windows) to ensure that game developers can seamlessly work with it, eliminating the need to switch to other APIs like OpenGL or Vulkan when porting games to Linux.

Note that OpenDX is currently in its early stages of development, with a massive [roadmap](https://github.com/EduApps-CDG/OpenDX/discussions/10) ahead. The current focus is on implementing DirectX 9 support, and there are plans to expand to support DirectX 12 in the future.

You can explore an [example](https://github.com/EduApps-CDG/OpenDX/blob/main/tests/basic_window.cpp) of the project where I am actively developing in the repository. You are welcome to test it by downloading the repository and [compiling the code](https://github.com/EduApps-CDG/OpenDX/blob/main/CONTRIBUTING.md) or by checking the [pre-releases](https://github.com/EduApps-CDG/OpenDX/releases) (which can have outdated code) if you want to.

So, in short. You still can't use it for playing games and dxdiag is the most advanced part of the project you can see visually.
